internal/setup/setup.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/nestoca/joy/internal/style"
+	"github.com/nestoca/joy/pkg/catalog"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+const defaultCatalogDir = "~/.joy"
+const separator = "————————————————————————————————————————————————————————————————————————————————"
+
+func Setup(configDir, catalogDir, catalogRepo string) error {
+	fmt.Println("👋 Hey there, let's kickstart your most joyful CD experience! ☀️")
+	fmt.Println(separator)
+
+	// Setup catalog and config
+	fmt.Print("🛠️ Let's first set up your configuration and catalog repo...\n\n")
+	catalogDir, err := setupCatalog(configDir, catalogDir, catalogRepo)
+	if err != nil {
+		return err
+	}
+	err = setupConfig(configDir, catalogDir)
+	if err != nil {
+		return err
+	}
+	fmt.Println(separator)
+
+	// Check dependencies
+	fmt.Print("🧐 Hmm, let's now see what dependencies you've got humming under the hood...\n\n")
+	checkDependencies()
+	fmt.Println(separator)
+
+	fmt.Println("🚀 All systems nominal. Houston, we're cleared for launch!")
+	return nil
+}
+
+func setupConfig(configDir string, catalogDir string) error {
+	// Try loading config file from given or default location
+	cfg, err := config.Load(configDir, catalogDir)
+	if err != nil {
+		return fmt.Errorf("loading config: %w", err)
+	}
+
+	// Save config file
+	cfg.CatalogDir = catalogDir
+	err = cfg.Save()
+	if err != nil {
+		return fmt.Errorf("saving config: %w", err)
+	}
+	fmt.Printf("✅ Saved config to file %s\n", style.Code(cfg.FilePath))
+	return nil
+}
+
+func setupCatalog(configDir string, catalogDir string, catalogRepo string) (string, error) {
+	var err error
+	catalogDir, err = getCatalogDir(configDir, catalogDir)
+	if err != nil {
+		return "", err
+	}
+
+	// Check if catalog directory exists
+	if _, err := os.Stat(catalogDir); err != nil {
+		if os.IsNotExist(err) {
+			err := cloneCatalog(catalogRepo, catalogDir)
+			if err != nil {
+				return "", err
+			}
+		} else {
+			return "", fmt.Errorf("checking for catalog directory %s: %w", catalogDir, err)
+		}
+	}
+
+	cat := loadCatalog(catalogDir)
+	printCatalogSummary(cat)
+	return catalogDir, nil
+}
+
+func getCatalogDir(configDir string, catalogDir string) (string, error) {
+	if catalogDir == "" {
+		// Try loading catalog dir from config file to use as prompt default value
+		cfg, err := config.Load(configDir, catalogDir)
+		if err == nil {
+			catalogDir = cfg.CatalogDir
+		} else {
+			catalogDir = defaultCatalogDir
+		}
+
+		// Prompt user for catalog directory using survey (defaults to $HOME/.joy)
+		err = survey.AskOne(&survey.Input{
+			Message: "🎯 Where does (or should) your local catalog reside?",
+			Help:    "This is where we will clone your catalog repo, but only if it's not already there.",
+			Default: catalogDir,
+		},
+			&catalogDir,
+			survey.WithValidator(survey.Required),
+		)
+		if err != nil {
+			return "", fmt.Errorf("prompting for catalog directory: %w", err)
+		}
+	}
+
+	// Expand tilde to home directory
+	homePrefix := "~/"
```

Comment:
There's a potential bug here, need a cleverer way to expand tilde:

`~riclima/.joy` is a valid way of writing `~/.joy`.

Maybe add a `// TODO` comment and make this a problem for a different day 🤷